authors/apps/articles/models.py

```diff
@@ -131,3 +128,32 @@ class Favorite(models.Model):
     """Implement storage of favorites"""
     user_id = models.ForeignKey('authentication.User', on_delete=models.CASCADE, related_name='favorites')
     article_id = models.ForeignKey('articles.articles', on_delete=models.CASCADE, related_name='favorites')
+
+
+class ReportArticles(TimeStampModel):
+    """ Model to hold instances of reports posted by users. """
+    PLAGIARISM = 'Plagiarism'
+    SPAM = 'Spam'
+    HARASSMENT = 'Harassment'
+    RULES_VIOLATION = 'Rules violation'
+    REPORT_CHOICES = (
+        (PLAGIARISM, 'Plagiarism'),
+        (SPAM, 'Spam'),
+        (HARASSMENT, 'Harassment'),
+        (RULES_VIOLATION, 'Rules violation'),
+    )
+    author = models.ForeignKey('authentication.User', related_name='reports', on_delete=models.CASCADE, null=False)
+    article =  models.ForeignKey('articles.Articles', related_name='reports', on_delete=models.CASCADE, null=False)
+    report = models.CharField(max_length=140)
+    type_of_report = models.CharField(
+        max_length=20,
```

Comment:
No need for `max_length` if you're making use of `choices`

Reply:
CharField requires max_length validation